### PR TITLE
Run `controlplane/transformation` integration tests in parallel

### LINKED ISSUE #124106: [Flaking Test] Integration transformation.transformation timeout 10m
### Which jobs are flaking?

- ci-kubernetes-integration-master

### Which tests are flaking?

- TestKMSv2Healthz https://prow.k8s.io/view/gs/kubernetes-jenkins/logs/ci-kubernetes-integration-master/1773469857552011264
- TestSecretsShouldBeTransformed https://prow.k8s.io/view/gs/kubernetes-jenkins/logs/ci-kubernetes-integration-master/1773515912264028160

### Since when has it been flaking?

since 03-28

https://storage.googleapis.com/k8s-triage/index.html?test=transformation found no failure.

### Testgrid link

https://testgrid.k8s.io/sig-release-master-blocking#integration-master

### Reason for failure (if possible)

```
panic: test timed out after 10m0s
running tests:
	TestKMSv2Healthz (2m42s)
```

```
panic: test timed out after 10m0s
running tests:
	TestSecretsShouldBeTransformed (6s)
```

### Anything else we need to know?

k8s.io/kubernetes/test/integration/controlplane/transformation.TestKMSv2Healthz(0xc003b464e0)
	/home/prow/go/src/k8s.io/kubernetes/test/integration/controlplane/transformation/kmsv2_transformation_test.go:893 +0x2f3

### Relevant SIG(s)

/sig testing auth
/cc @ritazh @aramase

### PR DESCRIPTION
**Pull request by aramase**
#### What type of PR is this?

/kind flake
/kind cleanup

#### What this PR does / why we need it:
- Generate a unique UDS path for each integration test, so there are no conflicts.
- Run some of the tests in parallel.

Test runtime before the change

```shell
➜ make test-integration WHAT=./test/integration/controlplane/transformation KUBE_TEST_ARGS="-v"
...
PASS
ok    k8s.io/kubernetes/test/integration/controlplane/transformation  502.481s
```

Runtime after change 

```shell
➜ make test-integration WHAT=./test/integration/controlplane/transformation KUBE_TEST_ARGS="-v"
...
PASS
ok  	k8s.io/kubernetes/test/integration/controlplane/transformation	375.110s
```

#### Which issue(s) this PR fixes:

Fixes https://github.com/kubernetes/kubernetes/issues/124106

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

/sig auth
/triage accepted
/priority important-longterm

### CODE REVIEW

**Comment by pacoxu** [comment]
- 9m22s in https://prow.k8s.io/view/gs/kubernetes-jenkins/pr-logs/pull/124250/pull-kubernetes-integration/1777885074008576000 this presubmit CI with this fix.
- 14m34s in CI without this change https://prow.k8s.io/view/gs/kubernetes-jenkins/logs/ci-kubernetes-integration-master/1782298252922589184

LGTM (non binding)

File: test/integration/controlplane/transformation/kms_transformation_test.go [hunk 1]
```diff
@@ -884,7 +891,10 @@ func TestEncryptionConfigHotReloadFilePolling(t *testing.T) {
 
 	for _, tc := range testCases {
 		t.Run(tc.name, func(t *testing.T) {
-			encryptionConfig := `
+			t.Parallel()
```
Comment (enj) [inline test/integration/controlplane/transformation/kms_transformation_test.go:894]: Need the `tc := tc` bit above.

**Review by enj — CHANGES_REQUESTED** [review]
Minor comments.

File: test/integration/controlplane/transformation/transformation_test.go [hunk 2]
```diff
@@ -638,3 +639,7 @@ func getLivez(checkName string, clientConfig *rest.Config, excludes ...string) (
 	body, err := req.DoRaw(context.TODO()) // we can still have a response body during an error case
 	return string(body), err == nil, nil
 }
+
+func getSocketPath() string {
+	return fmt.Sprintf("@%s.sock", rand.String(10))
```
[the remark this answers is not in the document]
Comment (aramase) [reply test/integration/controlplane/transformation/transformation_test.go:644]: the sockets are cleaned up here https://github.com/kubernetes/kubernetes/blob/d8f8c7fae0aecc5b72e1ab0f1b71d75833c6b40c/staging/src/k8s.io/apiserver/pkg/storage/value/encrypt/envelope/testing/v2/kms_plugin_mock.go#L146

**Review by neolit123 — COMMENTED** [review]
the bots pinged me on this PR for review. a checked the diff quickly but didn't comment.
today i logged this ticket due to flakes and remembered this PR:
https://github.com/kubernetes/kubernetes/issues/124476

**Pull request merged**